spaceapi/src/utils.rs

```diff
@@ -3,7 +3,7 @@
 /// An ``Optional`` can contain ``Optional::Value<T>`` or ``Optional::Absent``.
 /// It is similar to an ``Option``, but ``Optional::Absent`` means it will be
 /// omitted when serialized, while ``None`` will be serialized to ``null``.
-#[derive(Debug, Copy, Clone)]
+#[derive(Clone, Copy, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
```

Comment:
If you want to derive `Eq` you need to also derive `PartialEq`
```
src/utils.rs:6:35: 6:37 error: the trait `core::cmp::PartialEq` is not implemented for the type `utils::Optional<T>` [E0277]
src/utils.rs:6 #[derive(Clone, Copy, PartialOrd, Eq, Ord, Debug, Hash)]
                                                 ^~
src/utils.rs:6:35: 6:37 note: in expansion of #[derive_Eq]
```

See the doc-tests in `and_then` to see the comparison in works.